z3/src/optimize.rs

```diff
@@ -117,6 +149,34 @@ impl<'ctx> Optimize<'ctx> {
         Model::of_optimize(self)
     }
 
+    /// Retrieve the objectives for the last [`Optimize::check()`](#method.check)
+    ///
+    /// This contains maximize/minimize objectives and grouped soft constraints.
+    pub fn get_objectives(&self) -> Vec<Dynamic<'ctx>> {
+        let z3_objectives = unsafe {
+            let _guard = Z3_MUTEX.lock().unwrap();
+            Z3_optimize_get_objectives(self.ctx.z3_ctx, self.z3_opt)
+        };
+
+        let len = unsafe {
+            let _guard = Z3_MUTEX.lock().unwrap();
+            Z3_ast_vector_size(self.ctx.z3_ctx, z3_objectives)
+        };
```

Comment:
We could consolidate the unsafe blocks and lock acquisitions like:

```suggestion
        let (z3_objectives, len) = unsafe {
            let _guard = Z3_MUTEX.lock().unwrap();
            let objectives = Z3_optimize_get_objectives(self.ctx.z3_ctx, self.z3_opt);
            let len = Z3_ast_vector_size(self.ctx.z3_ctx, objectives);
            (objectives, len)
        };
```